submissions/xT0PERx/Friends app/js/index.js

```diff
@@ -0,0 +1,177 @@
+const allFriends = {
+  currentList: [],
+  changeList: [],
+  numberOfFriends: 72,
+};
+
+const REQUEST_URL = "https://randomuser.me/api/1.3/";
+const ALL_USERS_CARDS_FILTERS = allFriends.changeList;
+const ALL_USERS_CARDS = allFriends.currentList;
+const USERS_LIST = document.getElementById("usersList");
+const FILTERS_MENU = document.getElementById("filters_menu");
+const ABC_BUTTON_FILTER = document.getElementById("Abc");
+const ZYX_BUTTON_FILTER = document.getElementById("Zyx");
+const MALE_BUTTON_FILTER = document.getElementById("male");
+const FEMALE_BUTTON_FILTER = document.getElementById("female");
+const YOUNGER_FILTER_BUTTON = document.getElementById("younger-ftr");
+const SENIOR_FILTER_BUTTON = document.getElementById("senior-ftr");
+const RESET_BUTTON = document.getElementById("reset");
+const SEARCH_INPUT = document.querySelector(".searcher");
+
+function initApp() {
+  getUsers();
+}
+
+initApp();
+
+function getUsers() {
+  const url = `${REQUEST_URL}?results=${allFriends.numberOfFriends}`;
+  fetch(url)
+    .then((res) => {
+      if (res.ok) {
+        return res.json();
+      } else {
+        throw new Error("Что-то не так с сервером...");
+      }
+    })
+    .then((data) => {
+      saveUsers(data.results);
+      createUsersCards(ALL_USERS_CARDS);
```

Comment:
Single responsibility - your functions should do only one job. As an example function in which you fetch users should only fetch them and not render, transform or process them in other ways. All these things should be done in another place, outside your function

Reply:
Could you explain in more detail how I can process data without using functions in __fetch__. Perhaps I should save the data to a variable and then work with it, is this considered the best way?

Reply:
No, the global variable not the best way. Your get function should return the result of fetch, which is a promise with needed data and then you handle this promise as usual.